packages/@aws-cdk/pipelines/lib/docker-registry.ts

```diff
@@ -0,0 +1,164 @@
+import * as ec2 from '@aws-cdk/aws-ec2';
+import * as ecr from '@aws-cdk/aws-ecr';
+import * as iam from '@aws-cdk/aws-iam';
+import * as secretsmanager from '@aws-cdk/aws-secretsmanager';
+import { Fn } from '@aws-cdk/core';
+
+/** Docstrings */
+export abstract class DockerRegistry {
+  /** Docstrings */
+  public static fromDockerHub(opts: ExternalDockerRegistryOptions): DockerRegistry {
```

Comment:
To my mind, `from` is the wrong word here. It's either `forDockerHub`, or just `dockerHub` I'd say.

You're not making a `DockerRegistry`: you're describing one. Or the class should be called `DockerCredentials`...